File: src/main/java/org/example/model/UserInfo.java
package org.example.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "users")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class UserInfo {
    @Id
    private Long userId;
    private String chatId;
    private String username;
    private String firstName;
    private String lastName;

    @Column(name = "time_zone")
    private int timezone;
    private boolean feedbackModeAllowed;
    private boolean feedbackModeEnabled;
    private Long replyModeFolloweeId;
    private Integer replyModeMessageId;

    @OneToMany(mappedBy = "userInfo")
    private List<Subscription> subscriptions;

    @OneToMany(mappedBy = "followee")
    private List<Subscription> followers;


    public String getUserNameWithAt()
    {
        if (username != null)
        {
            return "@" + username;
        }
        if (lastName != null)
        {
            return firstName + " " + lastName;
        }
        return firstName;
    }
}

File: src/main/java/org/example/model/UserInfo.kt
package org.example.model

import java.util.*
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.OneToMany
import javax.persistence.Table

@Entity
@Table(name = "users")
class UserInfo {

    @Id
    var userId: Long? = null

    @Column(unique = true, nullable = false)
    var chatId: String? = null

    @Column(unique = true)
    var username: String? = null

    var firstName: String?= null

    var lastName: String?= null

    @Column(name = "time_zone")
    var timezone: Int?= null

    var feedbackModeAllowed: Boolean?= null

    var feedbackModeEnabled: Boolean?= null

    var replyModeFolloweeId: Long?= null

    var replyModeMessageId: Int?= null

    @OneToMany(targetEntity = UserInfo::class)
    var subscriptions: List<Subscription> = ArrayList()

    @OneToMany(targetEntity = Subscription::class)
    var followers: List<Subscription>?= ArrayList()

    fun getUserNameWithAt() : String? {
        if (username != null)
        {
            return "@$username"
        }
        if (lastName != null)
        {
            return "$firstName $lastName"
        }
        return firstName
    }
}